gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -175,6 +176,14 @@ class {{service.name}}RestTransport({{service.name}}Transport):
         # TODO(yon-mg): resolve other ctor params i.e. scopes, quota, etc.
         # TODO: When custom host (api_endpoint) is set, `scopes` must *also* be set on the
         # credentials object
+        maybe_url_match = re.match("^(?P<scheme>http(?:s)?://)?(?P<host>.*)$", host)
```

Comment:
Why do we need this? Looks like a heavyweight regexp to do validation. Isn't it safe to assume that if code reaches this poing the url already must be correct?

Reply:
It is not safe to make that assumption. The host is not validated at any point before this.

Reply:
right, but the host itself is not coming as a user input, but is rather generated, so it must be correct and the correctness of that is expected to be tested by relevant gapic-generator tests. Context: I'm just trying to avoid extra logic if possible on every request (anticipating Ads being unhappy with any extra millisecond spend on client layer).

Reply:
The host _may_ be passed in directly. There is a default, but it is overridable.
This logic happens once during transport construction, so I would be surprised if it shows up during their benchmark tests.